### PR TITLE
filter out non-anon questions from blocked/muted users in timelines

### LINKED ISSUE #585: filter out questions from people you're blocking in timelines
scenario:

```cucumber
Given as a fox
When i block user X
And i view my home timeline
Then i should not see any answers to user X's dogshit questions
```

### PR DESCRIPTION
**Pull request by nilsding**
~~just an experiment for now~~

attempt to resolve #585 -- please give it some thorough testing!

### CODE REVIEW

File: app/models/user/relationship/mute.rb [hunk 1]
```diff
@@ -16,19 +16,41 @@ module Mute
         has_many :muted_by_users, through: :passive_mute_relationships, source: :source
       end
 
+      # Mute an user
```
Comment (raccube) [inline app/models/user/relationship/mute.rb]:
```suggestion
      # Mute a user
```

File: app/models/user/relationship/block.rb [hunk 2]
```diff
@@ -43,6 +57,8 @@ def unfollow_and_remove(target_user)
         inboxes.joins(:question).where(questions: { user_id: target_user.id, author_is_anonymous: false }).destroy_all
         ListMember.joins(:list).where(list: { user_id: target_user.id }, user_id: id).destroy_all
       end
+
+      def cache_key_blocked_user_ids = "#{cache_key}/blocked_user_ids"
```
Comment (raccube) [inline app/models/user/relationship/block.rb:61]: Worth noting that `cache_key` includes the `updated_at` timestamp. Think it's fine here though.

**Review by raccube — APPROVED** [review]

File: app/models/user/relationship/block.rb (hunk 2 above)
Comment (nilsding) [reply app/models/user/relationship/block.rb:61]: yes, this is intentional

**Review by pixeldesu — APPROVED** [review]
LGTM 👍

**Pull request merged**